app/src/main/java/com/weatherxm/ui/home/quests/steps/GenericStepActivity.kt

```diff
@@ -0,0 +1,282 @@
+                        )
+                    )
+                ) {
+                    Box(
+                        modifier = Modifier
+                            .size(126.dp)
+                            .background(
+                                color = colorResource(R.color.colorSurface),
+                                shape = androidx.compose.foundation.shape.CircleShape
+                            ),
+                        contentAlignment = Alignment.Center
+                    ) {
+                        Icon(
+                            modifier = Modifier.size(48.dp),
+                            painter = painterResource(step.type.stepIcon()),
+                            contentDescription = null,
+                            tint = colorResource(R.color.colorPrimary)
+                        )
+                    }
+
+                    Column(
+                        verticalArrangement = Arrangement.spacedBy(
+                            dimensionResource(
+                                R.dimen.margin_extra_small
+                            )
+                        ),
+                        horizontalAlignment = Alignment.CenterHorizontally
+                    ) {
+                        Title(
+                            step.title,
+                            colorRes = R.color.textColor
+                        )
+                        MediumText(step.description)
+                    }
+
+                    if (step.isOptional) {
+                        Box(
+                            modifier = Modifier
+                                .background(
+                                    color = colorResource(R.color.colorSurface),
+                                    shape = androidx.compose.foundation.shape.CircleShape
+                                ),
+                            contentAlignment = Alignment.Center
+                        ) {
+                            MediumText(
+                                stringResource(R.string.optional_step_description),
+                                paddingValues = PaddingValues(
+                                    start = dimensionResource(R.dimen.padding_normal_to_large),
+                                    end = dimensionResource(R.dimen.padding_normal_to_large),
+                                    top = dimensionResource(R.dimen.padding_small),
+                                    bottom = dimensionResource(
+                                        R.dimen.padding_small
+                                    )
+                                )
+                            )
+                        }
+                    }
+                }
+            }
+            CtaButton(step, onCtaClick, onSkipClick)
+        }
+    }
+}
+
+@Composable
+private fun CtaButton(
+    step: QuestStep,
+    onCtaClick: () -> Unit,
+    onSkipClick: () -> Unit
+) {
+    Column(horizontalAlignment = Alignment.CenterHorizontally) {
```

Comment:
No need for this `Column`, this composable is already used inside another column with the correct alignment.